pmacApp/src/pmacController.cpp

```diff
@@ -1843,68 +1820,63 @@ asynStatus pmacController::fastUpdate(pmacCommandStore *sPtr) {
       debug(DEBUG_TRACE, functionName, "Real Time Interrupt %", rtPercent);
 
       // Background tasks percentage
-      if(Sys_BgSleepTime_ == 0)
-        bgSleepTime = 0.001;
-      else
-        bgSleepTime = (double)Sys_BgSleepTime_ / 1000000;
-        bgTaskTimeUs = bgTaskTimeUs - (double)(int)(bgTaskTimeUs/phaseDeltaTime)*phaseTaskTimeUs;
-        bgTaskTimeUs = bgTaskTimeUs - (double)(int)(bgTaskTimeUs/servoDeltaTime)*servoTaskTimeUs;
-        bgTaskTimeUs = bgTaskTimeUs - (double)(int)(bgTaskTimeUs/rtiDeltaTime)*rtTaskTimeUs;
-        bgTaskTime = bgTaskTimeUs / 1000000;
-        bgFreq = 1/ (bgSleepTime + bgDeltaTime/1000000);
-        bgPercent = (bgFreq * bgTaskTime)*100;
-        debug(DEBUG_TRACE, functionName, "Background Interrupt Frequency (Hz)", bgFreq);
-        debug(DEBUG_TRACE, functionName, "Background Interrupt Time (us)", bgTaskTimeUs);
-        debug(DEBUG_TRACE, functionName, "Background Interrupt %", bgPercent);
-
-      // Single-core Power PCservoDeltaTime
-      if (!strcmp(cpu_.c_str(), "PowerPC,460EX")) {
-        debug(DEBUG_TRACE, functionName, "CPU type", cpu_.c_str());
-        // Final CPU load calculation
-        cpuLoad = phasePercent + servoPercent + rtPercent + bgPercent;
-        debug(DEBUG_TRACE, functionName, "Calculated CPU %", cpuLoad);
-        setDoubleParam(PMAC_C_CpuUsage_, cpuLoad);
-      } else if ((!strcmp(cpu_.c_str(), "x86")) || (!strcmp(cpu_.c_str(), "PowerPC,APM86xxx")) || (!strcmp(cpu_.c_str(), "arm,LS1021A"))) {
-        // If CPU is dual core calculate the CPU load for CPU[1] (real time)
-        debug(DEBUG_TRACE, functionName, "CPU type", cpu_.c_str());
-        // Final CPU load calculation - Background tasks have dedicated core so calcs not actually used
-        cpuLoad = phasePercent + servoPercent + rtPercent;
-        debug(DEBUG_TRACE, functionName, "Calculated CPU %", cpuLoad);
-        setDoubleParam(PMAC_C_CpuUsage_, cpuLoad);
-      } else {
-        debugf(DEBUG_ERROR, functionName, "CPU [%s] not suppported", cpu_.c_str());
+      bgFreq = 1/ (bgDeltaTime/1000000);
+      bgTaskTimeUs = bgTimeUs     - (double)(int)(bgTimeUs/phaseDeltaTime)*phaseTaskTimeUs;
+      bgTaskTimeUs = bgTaskTimeUs - (double)(int)(bgTimeUs/servoDeltaTime)*servoTaskTimeUs;
+      bgTaskTimeUs = bgTaskTimeUs - (double)(int)(bgTimeUs/rtiDeltaTime)*rtTaskTimeUs;
+      bgPercent = (bgTaskTimeUs / bgDeltaTime)*100;
+      tasksPercent[PPMAC_CPU_BGTASK] = bgPercent;
+      debug(DEBUG_TRACE, functionName, "Background Interrupt Frequency (Hz)", bgFreq);
+      debug(DEBUG_TRACE, functionName, "Background Interrupt Time (us)", bgTaskTimeUs);
+      debug(DEBUG_TRACE, functionName, "Background Interrupt %", bgPercent);
+
+      for (int task_idx = 0; task_idx < PPMAC_CPU_TASKS_NUM; task_idx++) {
+        int core = cpuCoreTasks_[task_idx];
+        if (core >= 0) {
+          cpuLoad_[core] += tasksPercent[task_idx];
+        }
```

Comment:
cpuLoad_[core] is not initialised on line 1787 above
The if statement of 
`if(core >= 0) `
means that it is possible for an element of cpuLoad_ to remain uninitialised

The for loop below reads out from every element and so could try to read an uninitialised value

Reply:
cpuLoad_ initialised as vector.